### DISCUSSION TITLE
Rev 23 Modula-2 - Compiler

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#24] [General]
Was pleasantly surprised to find the Modula-2 compiler in <comdev>T_BUILD_TOOLS>MOD

The installer runs without error, yet even with a reboot

<comdev>T_BUILD_TOOLS>MOD>INSTALL_TEST*>MOD.INSTALL_TEST.CPL

shows the Copyright message but never completes. Just sits there twiddling thumbs.

(Tried as Sys Admin, regular user, and even from the console. Even tried  SPACing in case it was an obscure ACL issue.)

Curious to see how it compares to JPI/TopSpeed Modula-2 that I loved back-in-the-day.

```
[011] sman>pwd
<PIP01>T_BUILD_TOOLS>MOD>INSTALL_TEST*
[012] sman>r mod.install_test.cpl
nsed
mode nosemi
DEFINITION MODULE modtest;
PROCEDURE p(VAR x:short_integer);
END modtest.
file modtest.def.mod
INPUT

EDIT
mode nosemi

INPUT
DEFINITION MODULE modtest;
PROCEDURE p(VAR x:short_integer);
END modtest.

EDIT
file modtest.def.mod
File in use.  (TRNC$A)
nsed
mode nosemi
IMPLEMENTATION MODULE modtest;
PROCEDURE p(VAR x:short_integer);
BEGIN
x := 2;
END p;
BEGIN
END modtest.
file modtest.mod
INPUT

EDIT
mode nosemi

INPUT
IMPLEMENTATION MODULE modtest;
PROCEDURE p(VAR x:short_integer);
BEGIN
x := 2;
END p;
BEGIN
END modtest.

EDIT
file modtest.mod
nsed
mode nosemi
MODULE runmodtest;
From InOut   IMPORT WriteInt, WriteLn;
From modtest IMPORT p;
VAR i : SHORT_INTEGER;
BEGIN
   p(i);
   WriteInt(i,3);
   WriteLn;
END runmodtest.
file runmodtest.mod
INPUT

EDIT
mode nosemi

INPUT
MODULE runmodtest;
From InOut   IMPORT WriteInt, WriteLn;
From modtest IMPORT p;
VAR i : SHORT_INTEGER;
BEGIN
   p(i);
   WriteInt(i,3);
   WriteLn;
END runmodtest.

EDIT
file runmodtest.mod
   modula modtest.def
[MODULA Rev. T3.0-23.0 Copyright (c) 1986 Prime Computer, Inc.]
                                                                                
                                                                                
```

**Comment by SirNibble** [comment]
Should say, it responds to CTRL-P with a QUIT. and OK .

**Comment by drboone** [comment]
I can reproduce with both the 23.3 compiler and the 22.1 compiler.

I feel like we're chasing something underlying here.  First the BIND/PASCAL issue with file handles.  Then PL1 has a pointer/access issue writing to SYSPRINT.  Now MODULA hangs.

**Comment by SirNibble** [comment]
Just tried it on the 23.y2k example system (copied with magsav) with the same (lack of) result.

**Comment by SirNibble** [comment]
And if it helps ... results from the install test on Rev20

```
[MODULA Rev. T3.0-23.0 Copyright (c) 1986 Prime Computer, Inc.]

Error: condition "LINKAGE_ERROR$" raised at 4365(3)/24560.
"Not enough segments." while attempting
dynamic link to entrypoint "P$MAXF1TRF" .

```

**Reply by drboone** [reply]
This you can probably fix by just raising the number of dynamic segments in EDIT_PROFILE, then logging out and back in.

**Comment by hashbackup** [comment]
One thing to keep in mind is that mixing up software revs willy-nilly
was never supported "back in the day".  Even today, building software
on recent Linux systems, with new compilers and then trying to run it
on old systems with an old glibc isn't likely to work.

Just as a data point, when this Modula hang was first reported, I
connected to the emulator as a user and it hung for me too.  Then I
connected to the system console to run a usage and the CPU was pegged
with my user process.  Then I tried running f77 from the console and
it too hung, I'm assuming also in a CPU loop.

It could be an emulator problem, but the emulator is not actually
stuck.  It may have some kind of very subtle bug that is causing
Primos to get stuck in a fault-handling loop, or something along those
lines, because I couldn't ctrl-p out of the stuck CPU loop.  That can
also indicate that an OS lock is being held.

**Reply by SirNibble** [reply]
> One thing to keep in mind is that mixing up software revs willy-nilly was never supported "back in the day".

I'll keep that in mind, ta.

Obviously not really concerned about getting it to work on R20 as that just muddies the waters.

**Comment by drboone** [comment]
I'm wondering if the search rules on these assorted guests are sane, and intend to do a little digging along this line.  The _awesomesauce_ that results from the install scripts madly editing all over them is way uglier than I remember ours ever being.
